src/redux-middleware/pushQueue.ts

```diff
@@ -154,6 +163,44 @@ const flushDeletes =
     }
   }
 
+/** Pull children of pending merges and dispatch move to fully merge. */
```

Comment:
Maybe preface this with something like "When a thought is moved to another context, duplicates are recursively merged" to add more context.